rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -393,10 +427,8 @@ private NodeSet selectRecursively(int number, List<InetAddress> hostsSortedByNod
                         // found enough nodes on the same host
                         if (nodesOnHost.get(host).size() > number) {
                             // some extra nodes will be provided
-                            List<Node> nodes = nodesOnHost.get(host);
-                            NodeSet result = new NodeSet(nodes.subList(0, number));
-                            result.setExtraNodes(new LinkedList<>(nodes.subList(number, nodes.size())));
-                            return result;
+                            LinkedHashSet<Node> nodes = nodesOnHost.get(host);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "nodes" object should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)